### PR TITLE
1058/updated public listing filters

### PR DESCRIPTION
**Pull request by YazeedLoonat**
- Closes [#issue](https://github.com/CityOfDetroit/bloom/issues/1058)

## Description
This modifies the existing filters to match up with the mock in the linked issue

## How the filters work
**Availability filter:**
Vacant Units -> a unit group has availability, if paired with an open/closed waitlist filter option then the unit group must have availability and be marked as waitlist open/closed

open/closed waitlist -> a unit group must be marked as waitlist open/closed, if both are selected this filter does nothing

if a listing has a unit groups where at least one is marked as open waitlist and one is marked as closed waitlist that listing would always show regardless of open/closed/both waitlist options being selected as filters

**Bedroom Size**
this is functionally an or, so if you selected 2 different bedroom size options, a unit group must have at least one of those for the listing to show

**Monthly Rent Range**
this is an and functionality (if both are filled the rent must be between those values) 

**Community Programs**
this is functionally an or, so if you selected 2 different program options, at least one of those must be true on the listing for the listing to show

**Region**
This is functionally an or, if two or more are selected the listing must have at least one of those regions for the listing to show

**Accessibility Features**
This is functionally an and, if two are more are selected the listing must have all of those features for the listing to show

## How Can This Be Tested/Reviewed?
navigate to public site
navigate to the listings page
use the filter button, and verify the filter drawer matches with the mock
verify the filters work

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [x] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/public/src/forms/filters/FilterForm.tsx [hunk 1]
```diff
@@ -1,357 +1,268 @@
...
   filterState?: ListingFilterState
 }
 
+interface optionInterface {
+  value: string
+  label: string
+  translation?: string
+}
+
+const getTranslationString = (str: string) => {
```
Comment (emilyjablonski) [inline sites/public/src/forms/filters/FilterForm.tsx:32]: we could shorten this to something like `str.indexOf(“Bdrm”) ? str.replace (“Bdrm”, “Plus”) : str.push(“Plus)`

File: ui-components/src/sections/GridSection.tsx [hunk 2]
```diff
@@ -29,11 +29,13 @@ export interface GridSectionProps {
   reverse?: boolean
   children: React.ReactNode
   separator?: boolean
+  sectionClasses?: string
```
Comment (emilyjablonski) [inline ui-components/src/sections/GridSection.tsx:32]: This looks to be unused

**Comment by YazeedLoonat** [comment]
Hey @emilyjablonski this is ready for ya again!

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
Here's a round of visual QA for ya:

Styling updates 
* Section headers text (color, font, size)
* The sections themselves (need much less padding to match the mock)
* Footer styling (background color not reaching across, buttons misaligned)
* Section dividers (should go all the way across)
* Confirmed listings checkbox text (it is styled differently from all the other checkboxes, they should match and all be bold or all be light)
* Mobile styling (Doesn't match the designs, there is a mock that shows the textboxes vertical, as-is the checkbox sections are breaking on mobile views)
* Filter and show all buttons (the colors of the buttons aren't the same, the number styling is different and should be circular)

String updates
* `See All Listings` --> `Show all listings`
* `vacant units` --> `Vacant Units`
* `Open waitlist` -- > `Open Waitlist`
* `Closed waitlist` --> `Closed Waitlist`
* `Midtown - New Center` --> `Midtown`

Functional / design updates and questions
* I can't clear any filters
* The mock doesn't have SRO, are we supposed to have it?
* The community programs don't match the mock, should we have the same set?
* Being able to select both open and closed waitlist feels wrong since it seems additive. Selecting both does not show both open and closed waitlists, it shows nothing.

**Comment by YazeedLoonat** [comment]
Hey @emilyjablonski this should be ready for you to peep again

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
Some remaining changes from the last round:
* The footer background issue, the white background needs to reach all the way across and all the way down
* Footer styling (background color not reaching across and down)
* Show All Listings --> Show all listings

New notes:
* The drawer is not populating what I selected for the confirmed filter and any of the availability filters
* The waitlist filters don't seem to be playing nicely with the other filters. It seems like when I select a waitlist filter and another filter, nothing appears (ie you can try `Elevator` and `Open Waitlist` which should return results, and queries for them separately do, and some other combinations).

File: ui-components/src/icons/Icons.tsx [hunk 3]
```diff
@@ -415,6 +415,7 @@ export const Filter = (props: IconProps) => {
       viewBox="0 0 24 24"
       width="24px"
       fill={props.fill ?? "currentColor"}
+      className={props.className ?? ""}
```
Comment (emilyjablonski) [inline ui-components/src/icons/Icons.tsx:424]: Can't we just use the existing fill here?

File: sites/public/src/forms/filters/FilterForm.tsx [hunk 4]
```diff
@@ -300,8 +302,8 @@ const FilterForm = (props: FilterFormProps) => {
           />
         </GridCell>
       </GridSection>
-      <div className="text-left mt-8 bg-white border-t border-gray-450 pt-4">
-        <div className={"pb-4 mb-2 px-4"}>
+      <div className="text-left mt-28 bg-white border-t border-gray-450 pt-4">
```
Comment (emilyjablonski) [inline sites/public/src/forms/filters/FilterForm.tsx:305]: Screen Shot 2022-04-25 at 10 53 08 PM
A static margin like this will make the footer push to the bottom properly on a desktop view, but not elsewhere, where it will create an unnecessary gap.

**Comment by emilyjablonski** [comment]
From previous rounds:
* Still missing the circular number styling on the filter button
* Some of the changes made to the footer broke mobile a little (indicated in comment above)
* I don't think waitlist is working quite right. A simpler case,  I think Oak Village Independence should show for `Open Waitlist` and doesn't.

In the process of poking around at some of the bugs I was seeing I ended up with a bunch of fixes locally, and I just pushed em up and opened https://github.com/CityOfDetroit/bloom/pull/1183 lmk what you think!

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
Check out the comment above for changes, can you also fill out the checklist before merge? Thanks!

**Comment by YazeedLoonat** [comment]
Hey @emilyjablonski this is ready for your 👀  again, thank you very much for the help!

**Review by emilyjablonski — APPROVED** [review]
LGTM!
One existing issue I made a separate ticket for: https://github.com/bloom-housing/bloom/issues/2701

**Review by seanmalbert — APPROVED** [review]
